components/org.wso2.analytics.is.common.pdf/src/main/java/org/wso2/analytics/is/common/pdf/PDFGenerator.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.analytics.is.common.pdf;
+
+import org.pdfbox.exceptions.COSVisitorException;
+import org.pdfbox.pdmodel.PDDocument;
+import org.pdfbox.pdmodel.PDPage;
+import org.pdfbox.pdmodel.edit.PDPageContentStream;
+import org.pdfbox.pdmodel.graphics.xobject.PDJpeg;
+import org.pdfbox.pdmodel.graphics.xobject.PDXObjectImage;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PDFGenerator {
+    /*
+    *This method generates the PDF
+    *@param pdf Object of PDFPageInfo class
+    *@param table Object of Table class
+    *@param header Object of Header class
+    *@param footer Object of Footer class
+    *@param fileLocation Location of the file as a String
+    */
+    public void generatePDF(PDFPageInfo pdfPageInfo, Table table, Header header, Footer footer, String fileLocation) throws IOException, COSVisitorException {
+
+        File file = new File(fileLocation);
+        boolean fileCreated = false;
+        if (!file.exists()) {
+            fileCreated = file.createNewFile();
+        }
+        if(fileCreated) {
+            try(OutputStream os = new FileOutputStream(file)) {
+                renderPDF(pdfPageInfo, table, header, footer, os);
+            }
+        }
+    }
+
+    private void renderPDF(PDFPageInfo pdf, Table table, Header header, Footer footer, OutputStream outputStream) throws IOException, COSVisitorException {
+
+        if(table.getContent() != null && table.getContent().length != 0) {
+            PDDocument pdfDoc = null;
+            try {
+                pdfDoc = new PDDocument();
+                PDPage pdPage = generatePage(pdf, pdfDoc);
+                PDPageContentStream contentStream = drawHeader(header, pdfDoc, pdPage);
+                drawTable(pdfDoc, contentStream, pdf, table, footer);
+                pdfDoc.save(outputStream);
+            }
+            catch(Exception ex) {
```

Comment:
@DanojaDias 
Instead of just catching and throwing the exception can't we handle exact exceptions that could occur and log appropriately